business_objects/agreement.py

```diff
@@ -0,0 +1,37 @@
+from datetime import datetime
+from typing import Optional
+from submodules.model.session import session
+from submodules.model.business_objects import general
+from submodules.model.models import Agreement
+
+
+def get(project_id: str, item_id: str) -> Agreement:
+    return (
+        session.query(Agreement)
+        .filter(Agreement.project_id == project_id, Agreement.id == item_id)
+        .first()
+    )
+
+def get_by_xfkey(project_id: str, xfkey: str) -> Agreement:
+    return (
+        session.query(Agreement)
+        .filter(Agreement.project_id == project_id, Agreement.xfkey == xfkey)
+        .first()
+    )
+
+def create(
+    project_id: str,
+    user_id: str,
+    terms_text: str,
+    terms_accepted: bool,
+    xfkey: Optional[str] = None,
+    xftype: Optional[str] = None,
```

Comment:
how can these be optional? Maybe the key depending on the context but the type should always be filled

Reply:
It was build like this when the order of creation was not how it is now. We first created the agreement and then the embedding in the embedding service. Now it is other way around and we now can ensure that a key is present. For the type its the same i guess